### PR TITLE
Fix secret key from jwt 

### PR DESCRIPTION
**Pull request by AnthonyMed13**
Fix the secret key from jwt not being used correctly and added extra instructions on how to set up the .env file. Also added more error handling when it comes to the login and logout sessions to differentiate bewteen different problems. There should be no noticeable changes the login and logout, it should work correctly, although you may see some extra information been shown on the terminal.

### CODE REVIEW

File: backend/src/middleware/authenticateJWT.js [hunk 1]
```diff
@@ -11,15 +18,28 @@ const authenticateJWT = (req, res, next) => {
 
         jwt.verify(token, JWT_SECRET, (err, user) => {
             if (err) {
-                return res.sendStatus(403);  // Token invalid or expired
+                if (err.name === 'TokenExpiredError') {
+                    console.error('Error: Token has expired');
+                    return res.status(401).json({ message: 'Token expired. Please login again.' });
+                } else if (err.name === 'JsonWebTokenError') {
+                    console.error('Error: Invalid token');
+                    return res.status(401).json({ message: 'Invalid token. Authorization denied.' });
```
Comment (FrancesC8) [inline backend/src/middleware/authenticateJWT.js:26]: Should this case result in a 403? Same question for `TokenExpiredError`.

**Review by FrancesC8 — CHANGES_REQUESTED** [review]
Looks good, left minor comments.

File: backend/src/index.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ const usersHandler = new UsersHandler();
 // Create Controllers
 const healthCheckController = new HealthCheckController();
 const usersController = new UsersController(usersHandler);
-
+console.log('JWT_SECRET:', process.env.JWT_SECRET);
```
[the remark this answers is not in the document]
Comment (AnthonyMed13) [reply backend/src/index.js:32]: Done

File: backend/src/middleware/authenticateJWT.js [hunk 3]
```diff
@@ -1,6 +1,13 @@
 import jwt from 'jsonwebtoken';
+import crypto from 'crypto';
+import dotenv from 'dotenv';
+dotenv.config();
```
[the remark this answers is not in the document]
Comment (AnthonyMed13) [reply backend/src/middleware/authenticateJWT.js]: Done

File: backend/src/middleware/authenticateJWT.js (hunk 1 above)
Comment (AnthonyMed13) [reply backend/src/middleware/authenticateJWT.js:23]: For this case its if the user was not able to be authenticated which i think it should be 401. If I am not wrong 403 is even if the user was able to authenticate.

File: backend/src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (FrancesC8) [reply backend/src/index.js:40]: The `console.log('JWT_SECRET:', process.env.JWT_SECRET);` is still in the code.

[the remark this answers is not in the document]
Comment (AnthonyMed13) [reply backend/src/index.js]: Sorry ammended the wrong thing should be fine now

**Review by FrancesC8 — COMMENTED** [review]
Thanks!

**Review by FrancesC8 — APPROVED** [review]

**Review by AngelCIICMorales — CHANGES_REQUESTED** [review]
abroad@LAPTOP-54IVI1AE:/mnt/c/Users/Angel/Desktop/Experiment/semester-project-multiplayer-dominoes/backend$ npm run dev

> semester-project-multiplayer-dominoes-server@1.0.0 dev
> nodemon src/index.js

[nodemon] 3.1.6
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,cjs,json
[nodemon] starting `node src/index.js`
file:///C:/Users/Angel/Desktop/Experiment/semester-project-multiplayer-dominoes/backend/src/index.js:20
createMongoDbConnection(process.env.MONGODB_URI);
^

TypeError: Class constructor MongoDbConnection cannot be invoked without 'new'
    at file:///C:/Users/Angel/Desktop/Experiment/semester-project-multiplayer-dominoes/backend/src/index.js:20:1
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async loadESM (node:internal/process/esm_loader:28:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)

Node.js v20.11.0
[nodemon] app crashed - waiting for file changes before starting...

**Comment by AnthonyMed13** [comment]
> abroad@LAPTOP-54IVI1AE:/mnt/c/Users/Angel/Desktop/Experiment/semester-project-multiplayer-dominoes/backend$ npm run dev
> 
> > semester-project-multiplayer-dominoes-server@1.0.0 dev
> > nodemon src/index.js
> 
> [nodemon] 3.1.6 [nodemon] to restart at any time, enter `rs` [nodemon] watching path(s): _._ [nodemon] watching extensions: js,mjs,cjs,json [nodemon] starting `node src/index.js` file:///C:/Users/Angel/Desktop/Experiment/semester-project-multiplayer-dominoes/backend/src/index.js:20 createMongoDbConnection(process.env.MONGODB_URI); ^
> 
> TypeError: Class constructor MongoDbConnection cannot be invoked without 'new' at file:///C:/Users/Angel/Desktop/Experiment/semester-project-multiplayer-dominoes/backend/src/index.js:20:1 at ModuleJob.run (node:internal/modules/esm/module_job:218:25) at async ModuleLoader.import (node:internal/modules/esm/loader:329:24) at async loadESM (node:internal/process/esm_loader:28:7) at async handleMainPromise (node:internal/modules/run_main:113:12)
> 
> Node.js v20.11.0 [nodemon] app crashed - waiting for file changes before starting...

Fixed It was just a new change that was done to the index and I didn't merge them properly

**Review by AngelCIICMorales — APPROVED** [review]
LGTM

**Pull request merged**